packages/adapter-vercel/index.js

```diff
@@ -141,11 +156,44 @@ const plugin = function ({ external = [], edge, split } = {}) {
 
 			builder.log.minor('Writing routes...');
 
-			write(`${dir}/config.json`, JSON.stringify(config, null, '  '));
+			write(`${dir}/config.json`, JSON.stringify(static_config, null, '  '));
 		}
 	};
 };
 
+/**
+ * @param {import('.').Config | undefined} config_a
+ * @param {import('.').Config | undefined} config_b
+ */
+function can_group(config_a, config_b) {
+	if (config_a === config_b) return true;
+	if (!config_a || !config_b) return false;
+
+	if (config_a.runtime !== config_b.runtime) return false;
+	if (config_a.maxDuration !== config_b.maxDuration) return false;
+	if (config_a.memory !== config_b.memory) return false;
+	if (arrays_different(config_a.envVarsInUse, config_b.envVarsInUse)) return false;
```

Comment:
It kills me that you need to manually specify `envVarsInUse` — would love it if we could find a way to make this automatic, for example just always including anything specified in an `.env` file.

Regardless, I wonder if we should just combine all the `envVarsInUse` for a given group, so that we're not splitting more finely than we otherwise would?